rulesets/java/jpinpoint-rules.xml

```diff
@@ -1757,6 +1757,67 @@ class Foo {
         </example>
     </rule>
 
+    <rule name="InitializeComparatorOnlyOnce"
+          language="java"
+          message="Avoid creating Comparator instances repeatedly. Initialize them as static final fields instead."
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
+          externalInfoUrl="https://github.com/jborgers/PMD-jPinpoint-rules/tree/pmd7/docs/JavaCodePerformance.md">
+        <description>Problem: Creating Comparator instances repeatedly in methods like compareTo or sort calls is inefficient.&#13;
+            Solution: Initialize the Comparator as a static final field when not using external state and is thread safe.
```

Comment:
Removed thread safety note from doc, also remove here. I think indeed that it would be weird to put state in the comparator.

Reply:
Good catch

Reply:
I also see that the externalInfoUrl can use a anchor

Reply:
done